### PR TITLE
Add parameter for passing Jekyll configuration files

### PR DESCRIPTION
**Pull request by sbesson**
See https://trello.com/c/asX8jEBb/11-drop-wwwopenmicroscopyorg-baseurl

The scope of this PR is the addition of a new parameter in a backwards compatible manner as part of a minor release increment to control the configuration files.

The application of this new feature would be the case where one would want to deploy a website by combining multiple configuration files (see  https://github.com/openmicroscopy/www.openmicroscopy.org/pull/92). In the previous case, the consumer playbook would be changed to the following:

```
    - role: openmicroscopy.jekyll-build
      tags: jekyll
      jekyll_build_git_repo: "https://github.com/openmicroscopy/www.openmicroscopy.org"
      jekyll_build_force_rebuild: True
      jekyll_build_config: ["_config.yml" , "_prod.yml"]
```

A minimal test has been added consuming the branch matching the code PR mentioned above. The main caveat is that this puts a dependency of our infrastructure roles onto our code.

### CODE REVIEW

File: handlers/main.yml [hunk 1]
```diff
@@ -21,5 +21,7 @@
       --destination {{ jekyll_build_root }}
       {{ (jekyll_build_baseurl | length > 0) |
          ternary('--baseurl ' + jekyll_build_baseurl, '') }}
+      {{ (jekyll_build_config | length > 0) |
+         ternary('--config ' + jekyll_build_config, '') }}
```
Comment (manics) [inline handlers/main.yml]: Suggestion (not a blocker). Make `jekyll_build_config` a list of files and use the [`join` filter](http://docs.ansible.com/ansible/playbooks_filters.html#other-useful-filters)?

**Comment by sbesson** [comment]
Now that Jekyll installation on Docker has been fixed (https://github.com/sass/sass/issues/2345), the main remaining issues on this PR are:

- looking into using a list rather than a string https://github.com/openmicroscopy/ansible-role-jekyll-build/pull/5#discussion_r126946875. Does that modify the logic for checking the emptiness?
- fixing the failing test which is related to the default encoding on Docker. @manics: have you encountered this in the past. Is there any easy way to set UTF-8. I tried to use the `environment` option of the [docker driver](https://molecule.readthedocs.io/en/v1/driver/docker/index.html) to set LANG unsucessfully.

**Comment by manics** [comment]
> Does that modify the logic for checking the emptiness?

If the default is `[]` then `length > 0` should still work

> fixing the failing test which is related to the default encoding on Docker

I haven't had to deal with unicode in tests, but since it's python could you force the encoding with `sys.setdefaultencoding('utf8')` in `test_*.py`? https://stackoverflow.com/a/21190382

**Comment by manics** [comment]
```
TASK [ansible-role-jekyll-build : jekyll | clone site source code repository] ***
fatal: [jekyll-build-dual-config]: FAILED! => {"changed": false, "cmd": "/bin/git checkout --force prod_config", "failed": true, "msg": "Failed to checkout prod_config", "rc": 1, "stderr": "error: pathspec 'prod_config' did not match any file(s) known to git.\n", "stderr_lines": ["error: pathspec 'prod_config' did not match any file(s) known to git."], "stdout": "", "stdout_lines": []}
fatal: [jekyll-build-default]: FAILED! => {"changed": false, "cmd": "/bin/git checkout --force prod_config", "failed": true, "msg": "Failed to checkout prod_config", "rc": 1, "stderr": "error: pathspec 'prod_config' did not match any file(s) known to git.\n", "stderr_lines": ["error: pathspec 'prod_config' did not match any file(s) known to git."], "stdout": "", "stdout_lines": []}
```

Have you inadvertently hard-coded a dev branch?

**Pull request merged**